docs/community/developer-guide.md

```diff
@@ -206,47 +205,104 @@ matches the version of the Kubernetes server.
+official Microsoft extension titled "Rich Go language support for Visual Studio Code."
 
-Test individual package and individual test:
-```
-dlv test github.com/Azure/aks-engine/pkg/engine
-dlv test github.com/Azure/aks-engine/pkg/engine -- -test.run ^TestNetworkPolicyDefaults$
-```
+![install_go_extension.png](../static/img/install-go-extension.png)
 
-#### Visual Code Studio
+Once installed, the Go extension will `go get` several helper applications, including Delve for
+debugging support. You can read more about VS Code integration with Delve
+[here](https://github.com/Microsoft/vscode-go/wiki/Debugging-Go-code-using-VS-Code).
 
-More on VSC integration with delve can be found [here](https://github.com/Microsoft/vscode-go/wiki/Debugging-Go-code-using-VS-Code)
+Make sure you have the `aks-engine` code checked out to the appropriate location in your `$GOPATH`
+and open that directory in VS Code.
 
-If delve is installed and configured, you can use native VS functionality to debug code or individual tests (`debug test`)
+##### Debugging Tests
 
-Example launch.json file:
+If you are writing tests (and if you are, we want to give you a hug!), you can debug them directly
+in Visual Studio code.
+
+Set a breakpoint inside a test, then move your mouse pointer to the top of the function definition.
+To the right of "run test" appears a link saying "debug test": click it!
+
+![debug_test.png](../static/img/debug-test.png)
+
+##### Debugging AKS Engine
+
+To debug `aks-engine` itself, the default Go debugging configuration in `.vscode/launch.json` needs
+to be edited. Open that file (or just click the gear-shaped "Open launch.json" icon if you have the
+Debug panel open).
+
+Here is an example `launch.json` file that contains a configuration for debugging the
+`aks-engine generate` command using the `examples/kubernetes.json` file for its cluster
+configuration. Change the values in the `--set` arguments to reference your details.
 
 ```json
 {
   "version": "0.2.0",
   "configurations": [
     {
-      "name": "Launch",
+      "name": "aks-engine generate",
       "type": "go",
       "request": "launch",
       "mode": "debug",
       "program": "${workspaceRoot}",
       "env": {},
-      "args": ["generate", "${workspaceRoot}/examples/kubernetes.json"],
+      "args": [
+        "generate", "--api-model=${workspaceRoot}/examples/kubernetes.json",
+        "--set", "masterProfile.dnsPrefix=my-dns-prefix",
+        "--set", "linuxProfile.ssh.publicKeys[0].keyData=my-public-key-contents",
+        "--set", "servicePrincipalProfile.clientId=my-service-principal-client-id",
+        "--set", "servicePrincipalProfile.secret=my-service-principal-secret"
+      ],
       "showLog": true
     }
   ]
 }
 ```
 
+Copy and paste the configuration and change its name and details as needed. It's easy to build up
+a `launch.json` file to debug `aks-engine upgrade`, `scale`, and other commands and variations.
```

Comment:
Easy? For whom? What do you need to understand to make it easy?

Reply:
I meant this to be encouraging, but you're right that it's not "easy" in any objective sense. I'll rewrite this.